pkg/apis/stork/v1alpha1/schedulepolicy.go

```diff
@@ -0,0 +1,182 @@
+	"Wed":       time.Wednesday,
+	"Thursday":  time.Thursday,
+	"Thu":       time.Thursday,
+	"Thur":      time.Thursday,
+	"Thurs":     time.Thursday,
+	"Friday":    time.Friday,
+	"Fri":       time.Friday,
+	"Saturday":  time.Saturday,
+	"Sat":       time.Saturday,
+}
+
+// SchedulePolicyItem represents the schedule for executing an action
+type SchedulePolicyItem struct {
+	// Interval policy that will be triggered at the specified interval
+	Interval *IntervalPolicy `json:"interval"`
+	// Daily policy that will be triggered daily at a specified time
+	Daily *DailyPolicy `json:"daily"`
+	// Weekly policy that will be triggered on the specified day of the week at
+	// the specified time
+	Weekly *WeeklyPolicy `json:"weekly"`
+	// Monthly policy that will be triggered on the specified date of the month
+	// at the specified time
+	Monthly *MonthlyPolicy `json:"monthly"`
+}
+
+// IntervalPolicy contains the interval at which an action should be triggered
+type IntervalPolicy struct {
+	IntervalMinutes int `json:"intervalMinutes"`
+}
+
+// Validate validates an IntervalPolicy
+func (i *IntervalPolicy) Validate() error {
+	if i.IntervalMinutes < 1 {
+		return fmt.Errorf("Invalid intervalMinutes (%v) in Interval policy", i.IntervalMinutes)
+	}
+	return nil
+}
+
+// DailyPolicy contains the time in the day where an action should be executed
+type DailyPolicy struct {
+	Time string `json:"time"`
+}
+
+// GetHourMinute parses and return the hour and minute specified in the policy
+func (d *DailyPolicy) GetHourMinute() (int, int, error) {
+	return getHourMinute(d.Time)
+}
+
+// Validate validates a DailyPolicy
+func (d *DailyPolicy) Validate() error {
+	if _, _, err := d.GetHourMinute(); err != nil {
+		return fmt.Errorf("Invalid time (%v) in Daily policy: %v", d.Time, err)
+	}
+	return nil
+}
+
+func getHourMinute(policyTime string) (int, int, error) {
+	parsedTime, err := time.Parse(time.Kitchen, policyTime)
+	if err != nil {
+		parsedTime, err = time.Parse(strings.ToLower(time.Kitchen), policyTime)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+	return parsedTime.Hour(), parsedTime.Minute(), nil
+}
+
+// WeeklyPolicy contains the day and time in a week when an action should be
+// executed
+type WeeklyPolicy struct {
+	Day  string `json:"day"`
+	Time string `json:"time"`
+}
+
+// GetHourMinute parses and return the hour and minute specified in the policy
+func (w *WeeklyPolicy) GetHourMinute() (int, int, error) {
+	return getHourMinute(w.Time)
+}
+
+// Validate validates a WeeklyPolicy
+func (w *WeeklyPolicy) Validate() error {
+	if _, _, err := w.GetHourMinute(); err != nil {
+		return fmt.Errorf("Invalid time (%v) in Weekly policy: %v", w.Time, err)
+	}
+	if _, present := Days[w.Day]; !present {
+		return fmt.Errorf("Invalid day of the week (%v) in Weekly policy", w.Day)
+	}
+	return nil
+}
+
+// MonthlyPolicy contains the date and time in a month when an action should be
+// executed
+type MonthlyPolicy struct {
+	Date int    `json:"date"`
```

Comment:
Instead of `Date`, this should be called `Day`.  Cron follows this naming scheme: http://www.nncron.ru/help/EN/working/cron-format.htm

Reply:
Day or Date isn't part of Cron's scheme as such, they don't use any names.

The problem with using Day here is that it becomes confusing for a user to figure out that Day here is date of the month but in the WeeklyPolicy it is Day of the week.